src/statistics/means.pas

```diff
@@ -252,29 +255,91 @@ procedure TMeans.DoCalcAnova(ResultDF: TMeansDatafile);
         F := TEpiFloatField.DefaultMissing;
         BART := TEpiFloatField.DefaultMissing;
         PBART := TEpiFloatField.DefaultMissing;
+        W := TEpiFloatField.DefaultMissing;
+        PW := TEpiFloatField.DefaultMissing;
         exit; // Skip remaining statistics
       end
     else
       PROB := fdist(F, DFB, DFW);
   end;
 
-  // Should use Levene test of homogeneity of variances
-  // Bartlett's test is simpler, but is sensitive to non-normality of the data
+  k := lStrata + 1; // number of strata; lStrata fills in for k-1
+
+  // Levene's test of homogeneity of variances
+  // ref: https://www.itl.nist.gov/div898/handbook/eda/section3/eda35a.htm
+  // This is the 'medians' version of Levene's test
+  // Calculate Zi
+  with ResultDF do begin
+    setLength(zbar, k);
+    setLength(z, Obs);
+    idx := 0;
+    ytot := 0;
+    for i := 0 to lStrata do begin
+      ysum := 0;
+      m := Median.AsFloat[i];
+      for j := 0 to N.AsInteger[i] - 1 do begin
+        z[idx] := abs(CountVar.AsFloat[idx] - m);
+        ysum += z[idx];
+        inc(idx);
+      end;
+      ytot += ysum;
+      if (N.AsInteger[i] > 0) then
+        zbar[i] := ysum / N.AsFloat[i]
+      else
+        FExecutor.Error('Levene test error - n=1 in stratum ' + i.ToString);
```

Comment:
I'm uncertain, but doesn't this create a memory leak because ResultDF is never freed? The call to .Error(..) is raising an exception and it is not cause until we reach the executor. Hence the ResultDF (and all objects owned by it) will never be freed.